test/replutil.jl

```diff
@@ -336,3 +336,11 @@ withenv("JULIA_EDITOR" => nothing, "VISUAL" => nothing, "EDITOR" => nothing) do
     ENV["JULIA_EDITOR"] = "\"/Applications/Sublime Text.app/Contents/SharedSupport/bin/subl\" -w"
     @test Base.editor() == ["/Applications/Sublime Text.app/Contents/SharedSupport/bin/subl", "-w"]
 end
+
+# Issue #14684: `display` should prints associative types in full.
+let d = Dict(1 => 2, 3 => 45)
+    buf = IOBuffer()
+    td = TextDisplay(buf)
+    display(td, d)
+    @test bytestring(td.io) == "$(summary(d)):\n  3 => 45\n  1 => 2"
```

Comment:
It seems like this test depends on `Dict` values being printed in a particular order (but unlike https://github.com/JuliaLang/julia/pull/15853, it doesn't seem to matter now). Hashing and order might change in the future though.